Core/MIPS/RiscV/RiscVCompLoadStore.cpp

```diff
@@ -59,8 +59,19 @@ int32_t RiscVJitBackend::AdjustForAddressOffset(RiscVGen::RiscVReg *reg, int32_t
 		if (constant > 0)
 			constant &= Memory::MEMVIEW32_MASK;
 #endif
-		LI(SCRATCH2, constant);
-		ADD(SCRATCH1, *reg, SCRATCH2);
+		// It can't be this negative, must be a constant with top bit set.
+		if ((constant & 0xC0000000) == 0x80000000) {
```

Comment:
shouldn't this check be `(constant & 0x80000000)`? or are we explicitly diallowing 0xC0000000 addresses (granted, you did mention previously that disabling the cache in kernel space is likely not allowed...)

Reply:
No, because the constant can be a negative offset.  We end up turning absolute addresses into this immediate offsets, but we also have signed offsets.

-[Unknown]